internal/impl/kafka/output_sarama_kafka.go

```diff
@@ -296,6 +304,14 @@ func (k *kafkaWriter) buildUserDefinedHeaders(staticHeaders map[string]string) [
 
 //------------------------------------------------------------------------------
 
+// topicCache serves as a cache for topics that have been created during the
+// lifetime of this writer. This is to avoid unnecessary calls to the Kafka
+// cluster to create topics that already exist.
+//
+// The map is keyed by topic name and the value is a boolean indicating whether
+// the topic exists or not.
+var topicCache = syncmap.Map{}
```

Comment:
Two issues with this:

1. The memory is never reclaimed. In a long running service where inputs and outputs are spawned and torn down dynamically we would slowly grow this cache without any mechanism for clearing memory.
2. The cache being global means that multiple kafka outputs connecting to different clusters would conflict

Both issues can be solved by moving the cache into the `*kafkaWriter`. A config containing multiple kafka outputs would only double the number of look ups per topic which isn't a big deal.

Reply:
Yep, that's right, I overlooked this. I'll fix it right away